### PR TITLE
Upgrade to publishing components v19

### PR DESCRIPTION
**Pull request by injms**
## What
The upgrade to version 19 of the component gem will bring in the changes made to version 3 of GOV.UK Design System.

The legacy colours flag is turned on, and the legacy font size is set to be used. (There will be a separate pull request to remove this legacy code and to turn on the new colour palette.)

## Why
The improvements include a new focus state, which will help this app to reach the WCAG 2.1 standard. It also has a tonne of other improvements that cascade from Design System, such as the new smaller font.

## Visual regression results
https://government-frontend-pr-1448.surge.sh/gallery.html

## Component guide for this PR:
https://government-frontend-pr-1448.herokuapp.com/component-guide

### CODE REVIEW

File: app/assets/stylesheets/components/_download-link.scss [hunk 1]
```diff
@@ -4,9 +4,15 @@
   margin-top: govuk-spacing(3);
   margin-bottom: govuk-spacing(3);
 
+  &:focus {
+    @include govuk-focused-text;
+  }
+
```
Comment (chao-xian) [inline app/assets/stylesheets/components/_download-link.scss]: Super, super, picky but why are there 2 blank lines here?

Comment (maxgds) [reply app/assets/stylesheets/components/_download-link.scss]: Fixed

File: app/assets/stylesheets/components/_print-link.scss [hunk 2]
```diff
@@ -3,8 +3,9 @@
   margin-bottom: 2em;
 
   .app-c-print-link__link {
-    background: image-url("icon-print.png") no-repeat 0 50%;
-    padding: 0.5em 0 0.5em 28px;
+    background: image-url("icon-print.png") no-repeat 10px 50%;
+    padding: 0.5em 0.5em 0.5em 38px;
+    margin-left: -10px;
```
Comment (andysellick) [inline app/assets/stylesheets/components/_print-link.scss]: What's the reason for this change? It seems to be making the code more complicated.

File: app/assets/stylesheets/components/_published-dates.scss [hunk 3]
```diff
@@ -1,16 +1,26 @@
 .app-c-published-dates {
-  @include core-16;
```
Comment (andysellick) [inline app/assets/stylesheets/components/_published-dates.scss]: Again, you could try `govuk-link` on this element. I'm going to stop writing comments on this because there's a lot of them, the main benefit is that with the additional link class we're not duplicating the focus styles every time the mixin is called.

File: app/assets/stylesheets/components/helpers/_variables.scss [hunk 4]
```diff
@@ -20,6 +20,5 @@ $app-border-width-form-element: 2px;
 
 // Focus
 $app-focus-width: 3px;
-$app-focus-colour: $focus-colour;
 
 $app-error-colour: $red;
```
Comment (andysellick) [inline app/assets/stylesheets/components/helpers/_variables.scss]: Where does `$red` come from? Could we replace `$app-error-colour` with a govuk colour, directly in declarations rather than abstracting it out like this? Would make the code a bit less tunnelly.

File: app/assets/stylesheets/application.scss [hunk 5]
```diff
@@ -3,6 +3,11 @@
 $govuk-use-legacy-palette: true;
 $govuk-typography-use-rem: false;
 
+// govuk_frontend_toolkit dependencies
+@import "typography";
+@import "colours";
+@import "grid_layout";
+
```
Comment (andysellick) [inline app/assets/stylesheets/application.scss]: I'm confused by this commit - it doesn't include the bump to v19 as stated. Could it be merged into another commit?

File: app/assets/stylesheets/application.scss [hunk 6]
```diff
@@ -3,6 +3,11 @@
 $govuk-use-legacy-palette: true;
 $govuk-typography-use-rem: false;
 
+// govuk_frontend_toolkit dependencies
```
Comment (andysellick) [inline app/assets/stylesheets/application.scss]: Obviously a big ask, but do we still need these imports from the toolkit? This seems like a good opportunity to remove this dependency.

If it's a big job, can we look at it in a separate PR?

File: Gemfile [hunk 7]
```diff
@@ -28,6 +29,7 @@ group :development, :test do
 end
 
 group :development do
+  gem 'pry'
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply Gemfile]: This should be fine if it's in the development group.

File: app/assets/stylesheets/application.scss [hunk 8]
```diff
@@ -1,12 +1,15 @@
...
 @import 'govuk_publishing_components/all_components';
 
-// govuk_frontend_toolkit imports
-@import 'colours';
-@import 'shims';
-@import 'typography';
+// // govuk_frontend_toolkit imports
+// @import 'colours';
+// @import 'shims';
+// @import 'typography';
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply app/assets/stylesheets/application.scss]: It's deleted in a later commit, and I did feedback f2f yesterday.

File: Gemfile.lock [hunk 9]
```diff
@@ -389,4 +392,4 @@ RUBY VERSION
    ruby 2.6.3p62
 
 BUNDLED WITH
-   1.17.3
+   2.0.2
```
Comment (vanitabarrett) [inline Gemfile.lock]: Should we be committing this?

File: app/assets/stylesheets/components/_important-metadata.scss [hunk 10]
```diff
@@ -30,4 +29,8 @@
 
 .app-c-important-metadata__definition {
   font-weight: bold;
+
+  a:focus {
```
Comment (vanitabarrett) [inline app/assets/stylesheets/components/_important-metadata.scss]: I think we can use the `govuk-link` class instead?

File: Gemfile (hunk 7 above)
[the remark this answers is not in the document]
Comment (maxgds) [reply Gemfile]: Ian found it useful and introduced me to it recently. Could remove it but it'd mean putting it back in every time you want to debug something using it so I think it's worth leaving in.

File: app/assets/stylesheets/components/_print-link.scss (hunk 2 above)
Comment (maxgds) [reply app/assets/stylesheets/components/_print-link.scss]: It was to give breathing space to the focus stage on the print icon and bring it in line with the next/previous buttons it sometimes sits by. This is the current state:

Screen Shot 2019-09-04 at 13 57 04

File: app/assets/stylesheets/application.scss (hunk 6 above)
Comment (maxgds) [reply app/assets/stylesheets/application.scss]: The tickets for this work are all "do the minimal to bring in the V3 code" with an extra ticket for each to do the tidying up of that kind. The reason being we need to get all the frontend apps into a stable state so that people can adopt new components/update compoents they're using the the apps. The main scope for us of this work is to get the focus states up to date - all the other stuff is nice-to-have but not strictly part of our mission and the progress we are making is really slow, so no, this can't be done immediately but we hope to get to a point where we can.

File: app/assets/stylesheets/application.scss (hunk 5 above)
Comment (maxgds) [reply app/assets/stylesheets/application.scss]: Damnit... that's down to the rebase from master I had to do this morning. Will fix.

File: app/assets/stylesheets/components/helpers/_variables.scss (hunk 4 above)
Comment (maxgds) [reply app/assets/stylesheets/components/helpers/_variables.scss]: No idea, but we didn't introduce this in this PR.

File: Gemfile.lock (hunk 9 above)
Comment (maxgds) [reply Gemfile.lock]: It's reverted later, it was a mistake.

File: test/wraith/config.yaml [hunk 11]
```diff
@@ -35,7 +35,7 @@ domains:
 paths:
   # Most visited Answers (19 April 2017)
   answers_0: '/national-minimum-wage-rates'
-  answers_1: '/apply-uk-visa'
+  answers_1: '/apply-to-come-to-the-uk'
```
[the remark this answers is not in the document]
Comment (maxgds) [reply test/wraith/config.yaml]: fixed

File: app/assets/stylesheets/components/_back-to-top.scss [hunk 12]
```diff
@@ -1,9 +1,12 @@
 .app-c-back-to-top {
-  @include core-16;
   display: inline-block;
```
[the remark this answers is not in the document]
Comment (maxgds) [reply app/assets/stylesheets/components/_back-to-top.scss]: So I tried the govuk-link class only and got this:
Screen Shot 2019-09-04 at 14 55 19

The black text is lost unless I explicity set it. Happy to take input on how to proceed. I guess I could declare it for govuk-link:focus so it's not so repetitive? Suggestions?

[the remark this answers is not in the document]
Comment (maxgds) [reply app/assets/stylesheets/components/_back-to-top.scss]: As discussed, will do this in a separate pr

File: app/assets/stylesheets/application.scss (hunk 5 above)
Comment (maxgds) [reply app/assets/stylesheets/application.scss]: Fixed with a squash, rebase confused things.

File: app/assets/stylesheets/components/_important-metadata.scss (hunk 10 above)
Comment (maxgds) [reply app/assets/stylesheets/components/_important-metadata.scss]: Will do in a separate pr as discussed with Andy
